doc/source/whatsnew/v0.24.0.txt

```diff
@@ -314,7 +314,7 @@ Deprecations
 - :meth:`DataFrame.to_stata`, :meth:`read_stata`, :class:`StataReader` and :class:`StataWriter` have deprecated the ``encoding`` argument.  The encoding of a Stata dta file is determined by the file type and cannot be changed (:issue:`21244`).
 - :meth:`MultiIndex.to_hierarchical` is deprecated and will be removed in a future version  (:issue:`21613`)
 - :meth:`Series.ptp` is deprecated. Use ``numpy.ptp`` instead (:issue:`21614`)
--
+- :meth:`Series.compress` is deprecated. Use ``Series.__getitem__`` instead (:issue:`18262`)
```

Comment:
I prefer what you said in the warning message, which is to use `Series[condition]` instead (more end-user friendly IMO vs. `Series.__getitem__`).